src/compiler/emitter.ts

```diff
@@ -320,7 +320,7 @@ var __param = (this && this.__param) || function (paramIndex, decorator) {
 
         const awaiterHelper = `
 var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
-    return new P(function (resolve, reject) {
+    return new (P || (P = Promise))(function (resolve, reject) {

```

Comment:
We are leaving support for custom Promise return types for ES5 and earlier which do not have a global Promise, so we want to keep the same argument list. The default behavior now will be to use the Promise captured in the same scope as `__awaiter`. This is to avoid issues when creating closures in namespaces:

```ts
// input.ts
export namespace x {
  var Promise = "Not a Promise constructor.";
  async function f(): Promise { // references global Promise type
  }
}

// output.js
var __awaiter = ...; // captures global Promise here
export var x = (function (x) {
  var Promise = "Not a Promise constructor."; // local Promise var shadows global Promise
  function f() {
    // Before this change, we would capture Promise here, which would be the wrong Promise:
    // return __awaiter(this, void 0, Promise, function *() { });
    
    // After this change, we use the Promise captured at the top-level of the module.
    return __awaiter(this, void 0, void 0, function *() { });
  }
})(x || (x = {}));
```